packages/shared/src/components/post/PostEngagements.tsx

```diff
@@ -82,6 +84,31 @@ function PostEngagements({
     }
   }, [shouldOnboardAuthor]);
 
+  const hasCommentToggleQueryParam =
+    typeof router.query.comment !== 'undefined';
+
+  useEffect(() => {
```

Comment:
Just this part I think we have a hook now for interacting with query params.
Would prefer to use that to keep everything inline.

And please check with @sshanzel on the actual implementation.
Really lke the suggestion here though.

Reply:
Refactored inside `usePostComment` here bea2d8438dc0de12aad077cbb8f5863018d5f225